src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectKey.cs

```diff
@@ -46,4 +45,13 @@ public bool Equals(ProjectKey other)
     {
         return FilePathComparer.Instance.Equals(Id, other.Id);
     }
+
+    /// <summary>
+    /// Returns <see langword="true"/> if this <see cref="ProjectKey"/> matches the given
+    /// <see cref="Project"/>.
+    /// </summary>
+    public bool Matches(Project project)
+    {
+        return FilePathNormalizer.AreDirectoryPathsEquivalent(Id, project.CompilationOutputInfo.AssemblyPath);
```

Comment:
This code _looks_ really suspect, as its comparing a folder to a filename, and based on the tests I assume it only works because `Id` will always end in a `/`. It might be good to add a comment to that effect, and perhaps even Debug.Assert that it is true.

Reply:
You are correct sir - good eye! Great suggestion to add an Assert here. I will do so.